src/test/java/com/library/controller/BookControllerTest.java

```diff
@@ -0,0 +1,207 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@WebMvcTest(BookController.class)
+@Import(value = {BookServiceImpl.class, ModelMapper.class, BookMapper.class, TestStorage.class})
+@AutoConfigureMockMvc
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private BookService bookService;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private BookMapper bookMapper;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    @Autowired
+    private TestStorage  testStorage;
+
+    private static final String MISSING_TYPE = "source cannot be null";
+
+    private static final String DELETED_SUCCESS = "Book is deleted successfully!";
+
+    @SneakyThrows
+    @Test
+    void getAllBooks() {
+        Pageable paging = PageRequest.of(0, 5);
+        when(bookRepository.findAll(paging)).thenReturn(new PageImpl<>(testStorage.getBooks()));
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books")
+                        .param("page", "1")
+                        .param("sizeOnPage", "5"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+        JSONArray jsonArray = new JSONArray(response.getContentAsString());
+        JSONObject jsonObject = jsonArray.getJSONObject(0);
+
+        assertNotNull(response.getContentAsString());
+        assertEquals(1, jsonObject.get("id"));
+        assertEquals("War and peace", jsonObject.get("title"));
+        assertEquals(3, jsonArray.length());
+        assertEquals("application/json", response.getContentType());
+        verify(bookRepository).findAll(paging);
+    }
+
+    @SneakyThrows
+    @Test
+    void getBookById() {
+        when(bookRepository.findById(1L)).thenReturn(Optional.of(testStorage.getBook()));
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books/{id}", "1"))
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andReturn()
+                .getResponse();
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertEquals("update", jsonObject.get("title"));
+        assertEquals(1, jsonObject.get("id"));
+        verify(bookRepository).findById(1L);
+    }
+
+    @SneakyThrows
+    @Test
+    void getBookByIdFailed() {
+        when(bookRepository.findById(4L)).thenThrow(EntityNotFoundException.class);
+
+        MockHttpServletResponse response = mockMvc.perform(get("/books/{id}", "4"))
+                .andDo(print())
+                .andExpect(status().isNotFound())
+                .andReturn()
+                .getResponse();
+        JSONObject jsonObject = new JSONObject(response.getContentAsString());
+
+        assertEquals("null", jsonObject.get("message").toString());
```

Comment:
Some tests have verifying of mocks some does not.... be carefull and write test in same style.
Please check all tests.

Reply:
ok